uboot/secure_boot/imx8mm_habv4.rst

```diff
@@ -0,0 +1,594 @@
+and certificates.
+
+i.MX8M Boot Flow
+================
+
+The boot flow on i.MX 8M family devices are slightly different when compared with i.MX 6, i.MX 7
+series device. The boot flow changes are due to the different architecture, multiple firmware and
+software components required to initialize the device[I].
+
+.. image:: /images/imx8mm_bootflow.png
+Figure III: i.MX8M Boot Flow[I] 
+
+ - NOTE: for i.MX8M mini hdmi/dp firmware is not required in the boot process
+
+On reset, ROM code reads the efuse to determine the security configuration of the SoC and
+the type of the boot device. The ROM then locates flash.bin, layout shown in Figure IV.
+
+.. image:: /images/imx8mm_flash.png
+Figure IV: i.MX8M Boot Flow[I] 
+
+The above Figure shows 2 CSF binaries required, one per verification stage:
+
+ - CSF SPL that is used by Boot ROM to authenticate SPL + DDR FW
+   
+ - CSF FIT that is used by SPL (through HAB APIs) to authenticate FIT components
+
+The ROM code first reads the Image Vector Table(IVT) and identifies the CSF location. It verifies
+and loads Second Program Loader (SPL) and DDR firmware onto OCRAM. It verifies the authenticity of
+the SPL binary both in memory and flash against the signature embedded in the CSF. This can be seen
+as the first stage verification. If signature verification fails, execution is not allowed to leave
+the ROM for securely configured SoCs, also called “closed” devices.
+
+For an open device (meaning the fuse related to configuring SOC is not yet programmed), even if signature
+verification fails one will be able to see HAB events which will tell the programmer if the image would
+pass the authentication process.
+
+During Verification, HAB evaluates the SRK table included in the signature by hashing it and comparing
+the result to the SRK fuse values. If the SRK verification is successful, this establishes the root of
+trust, and the remainder of the signature can be processed to authenticate the image.
+
+If it succeeds (or if the device is open), the ROM code goes loads and jumps to SPL. SPL initializes the
+clocks, PMIC, LPDDR4 and UART. SPl then loads and verifies the rest of the components using the HAB APIs
+to extend the root of trust.
+
+A FIT image is used to package all the components rtequired for i.MX8M mini which are:
+
+ - U-Boot binary + device tree blob (dtb)
+
+ - ARM Trusted Firmware (ATF)
+   
+ - OP-TEE binary (TrustZone - optional)
+
+Once the DDR is available, the SPL code verifies and loads all the images included in the FIT structure to
+their specific execution addresses. This is the second stage and if the verification succeeds, ATF is
+executed which will in turn execute U-Boot.
+
+The HABv4 APIs can be used to extend the root of trust additionally tothe Linux Kernel and Cortex-M images
+and can be authenticated at bootloader level.
+
+
+The below procedure outlines on how signed secure boot has been created with Engicam i.CoreMX8M MINI CTouch2.0 board.
+
```

Comment:
Please create a flow chart for creating singed image steps? like
https://wiki.amarulasolutions.com/uboot/secure_boot/imx6_habv4.html#image-signer-flow